Jellyfin.Server/Middleware/UrlDecodeQueryFeature.cs

```diff
@@ -52,18 +52,8 @@ public IQueryCollection Query
                     return;
                 }
 
-                // Unencode and re-parse querystring.
-                var unencodedKey = HttpUtility.UrlDecode(key);
-
-                if (string.Equals(unencodedKey, key, StringComparison.Ordinal))
-                {
-                    // Don't do anything if it's not encoded.
-                    _store = value;
-                    return;
-                }
-
                 var pairs = new Dictionary<string, StringValues>();
-                var queryString = unencodedKey.SpanSplit('&');
+                var queryString = HttpUtility.UrlDecode(key).SpanSplit('&');
```

Comment:
You're wasting an allocation (dictionary) here if there's no ampersand in the query string

Reply:
Don't think this is true, if we only encoded a single value there wouldn't be an `&`We need to check for an `=`, if there isn't one we can return early